### PR TITLE
Release 4.4.0

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

**Comment by hardbyte** [comment]
I've switched out a part of the publishing pipeline so keen to try a test release to see if the release candidate actually gets published.

**Comment by hardbyte** [comment]
Pre-release (after 4 attempts!) made it to PyPI:

```
pip install python-can==4.3.2rc4
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
+Version 4.3.2
+=============
+
+Features
+--------
+
+* TRC 1.3 Support: Added support for .trc log files as generated by PCAN Explorer v5 and other tools, expanding compatibility with common log file formats (#1753).
+* ASCReader Performance: Significantly improved the read performance of ASCReader, optimizing log file processing and analysis (#1717).
```
Comment (zariiii9003) [inline CHANGELOG.md:8]: It's probably not much of a difference in performance. The code is a little bit cleaner though.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+* SYSTEC Interface Enhancements: Added the ability to pass an explicit DLC value to the send() method when using the SYSTEC interface, enhancing flexibility for message definitions (#1756).
+* Socketcand Beacon Detection: Introduced a feature for detecting socketcand beacons, facilitating easier connection and configuration with socketcand servers (#1687).
+* PCAN Driver Echo Frames: Enabled echo frames in the PCAN driver when receive_own_messages is set, improving feedback for message transmissions (#1723).
+* CAN FD Bus Connection: Enabled connecting to CAN FD buses without specifying bus timings, simplifying the connection process for users (#1716).
```
Comment (zariiii9003) [inline CHANGELOG.md:12]: This only affects VectorBus, we should mention that.

**Review by zariiii9003 — APPROVED** [review]
Nice effort, looks good. I'd prefer 4.4.0 though since these are not just bug fixes and might introduce new bugs.

File: doc/development.rst [hunk 3]
```diff
@@ -117,13 +117,19 @@ Creating a new Release
...
   `Read the Docs <https://readthedocs.org/projects/python-can/versions/>`__ and
   `GitHub <https://github.com/hardbyte/python-can/releases>`__.
+
+
+Manual release steps (deprecated)
+---------------------------------
+
+- Create a temporary virtual environment.
+- Build with  ``pipx run build``
+- Create and upload the distribution: ``python setup.py sdist bdist_wheel``.
```
Comment (zariiii9003) [inline doc/development.rst:133]: This is redundant, `pipx run build` creates wheel and sdist

**Pull request merged**